File: baekjoon/12018.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int numberOfSubject, mileage, p, l, count = 0;
    cin >> numberOfSubject >> mileage;
    vector<int> pList;
    vector<int> answer;

    for (int i = 0; i < numberOfSubject; i++)
    {
        cin >> p >> l;
        pList.resize(p);
        for (int i = 0; i < p; i++)
        {
            cin >> pList[i];
        }
        if (p < l)
        {
            answer.push_back(1);
        } else
        {
            sort(pList.begin(), pList.end());
            answer.push_back(pList[p-l]);
        }
    }
    sort(answer.begin(), answer.end());
	int cnt = 0;
	for (int i : answer)
	{
		if (mileage >= i)
		{
			count++;
			mileage -= i;
		}
	}
    cout << count;
    return 0;
}

File: baekjoon/12018.py
import sys
n, m = map(int, sys.stdin.readline().split())
answer = []
cnt = 0 
for _ in range(n):
    p, l = map(int, sys.stdin.readline().split())
    p_list = sorted(list(map(int, sys.stdin.readline().split())))
    if p >= l:
        answer.append(p_list[l-2])
    else:
        answer.append(1)
answer = sorted(answer)
for i in range(m):
    m -= answer[i]
    if m < 0:
        break
    cnt += 1
print(cnt)
